code/modules/events/grey_tide.dm

```diff
@@ -26,26 +27,29 @@
 	)
 
 	for(var/i in 1 to severity)
-		var/picked_area = pick_n_take(potential_areas)
-		for(var/area/area_to_check as anything in GLOB.areas)
-			if(istype(area_to_check, picked_area))
-				areas_to_open += area_to_check
+		GLOB.grey_tide_areas += pick_n_take(potential_areas)
 
 /datum/round_event/grey_tide/announce(fake)
 	priority_announce("Gr3y.T1d3 virus detected in [station_name()] secure locking encryption subroutines. Severity level of [severity]. Recommend station AI involvement.", "Security Alert")
 
+	GLOB.grey_tide_areas.Cut() //As announce always occurs last, we use it to clean up the list of areas to be affected
+
 /datum/round_event/grey_tide/start()
-	if(!length(areas_to_open))
+	if(!length(GLOB.grey_tide_areas))
 		log_world("ERROR: Could not initiate grey-tide. No areas in the list!")
 		kill()
 
 /datum/round_event/grey_tide/tick()
 	if(ISMULTIPLE(activeFor, 12))
-		for(var/area/area_to_open in areas_to_open)
+		for(var/area/area_to_open in GLOB.grey_tide_areas) //Currently broken due to the changes to areas_to_open. Doesn't matter because this functionality was getting thrown out anyways
```

Comment:
Can you describe this more in detail in the comment? What is the problem and when are we removing this? Or is this your TODO?